### ISSUE TITLE
Add High DPI (4K) Menu's and Subtitles

### ISSUE THREAD

**Issue by Michael-Z-Freeman** [ValveSoftware/source-sdk-2013#530]
Source SDK 2013 mods all scale the in game HUD elements well to 4K (I refer to latest beta version). The exception is subtitles/messages although that can be fixed by altering the correct ".res" file to show a larger font size. For example see - https://developer.valvesoftware.com/wiki/Modifying_Source_GameUI - I have confirmed by testing it that fonts can in fact be increased in size to be visible on my 4K display. The settings menu does not scale to 4K but could also be fixed although this is a bigger job. It uses VGUI - https://developer.valvesoftware.com/wiki/VGUI_Documentation - There are even "high_def" switches in the resource files to allow different sizes for different displays. I experimented with modding this myself but its quite a big job for someone not familiar with VGUI as an entire menu layout needs adding for 4K (3840x2160). But its obviously possible. So I don't understand why Valve have not updated their SDK to be fully visible on 4K/High DPI displays by default (default settings menu and subtitle sizes). Half Life 2 mods look really good in 4K but I suspect most people don't stay with it for very long because the player cannot see Keyboard controls to change or other settings if they try and go into the menu in 4K. It just becomes a massive inconvenience so I usually end up dropping down to 1920x1080. So it would be nice if the SDK were updated for modern displays.

**Comment by schm1dtHL** [comment]
The solution should probably be to stop using any non-proportional fonts or UI elements in code, the hidef stuff you refer to was only for the Xbox 360 version of TOB in order to support 720p and 480p simultaneously with correctly sized UI elements for each, without having to switch everything to be proportional. I would think that Valve intends to replace the current menu system for PC at some point though with the gamepad UI that is used by the Steam Deck by default(or with the -gamepadui launch option), which automatically scales the UI correctly anyways.

**Comment by Michael-Z-Freeman** [comment]
> The solution should probably be to stop using any non-proportional fonts or UI elements in code, that way the UI would ideally not need to be adapted for a million different resolutions. The hidef stuff you refer to was only for the Xbox 360 version of TOB in order to support 720p and 480p simultaneously with correctly sized UI elements for each res, without having to switch everything to be proportional. I would think that Valve intends to replace the current menu system for PC HL2/EP1/EP2 at some point though with the fancy new gamepad UI that is used by the Steam Deck by default(or with the -gamepadui launch option elsewhere), which automatically scales the UI correctly at different resolutions.

Yes I just checked and the beta has the new UI and it is 4K ready !

https://www.gamingonlinux.com/2022/01/half-life-2-getting-a-new-ui-in-prep-for-steam-deck/

You have to use "-gamepadui" as a launch option.

It's a beta so I had to fiddle around a little to get it working in 4K. Changing the resolution leads to an update problem with the new UI which is fixed by restarting. 

Here's a nice comparison between old and new - https://www.pcgamesn.com/steam-deck/half-life-2-ui-updat

![Half-Life 2_2022 01 30-15 52](https://user-images.githubusercontent.com/951566/151707243-3f57de20-e9df-48b1-a591-e9b386ff0597.jpg)
![Half-Life 2_2022 01 30-15 52_1](https://user-images.githubusercontent.com/951566/151707246-27d60a85-6a9b-48de-9bdb-1133a392313c.jpg)
![Half-Life 2_2022 01 30-15 53](https://user-images.githubusercontent.com/951566/151707247-809b813a-43c7-4948-bb7a-ef480549fc12.jpg)
![Half-Life 2_2022 01 30-15 53_1](https://user-images.githubusercontent.com/951566/151707249-83e627e2-be48-4a05-b8f7-13ba25758ea6.jpg)
e

**Comment by AL2009man** [comment]
Seems like Left 4 Dead 2 and Portal 2 also added "-gamepadui" function, the effects aren't as drastic as HL2's.

with Left 4 Dead 2, it only affects the HUD:

Before:
![20220201154038_1](https://user-images.githubusercontent.com/67606569/153130137-c8f7809b-9f87-402b-93b7-4fc7931cf934.jpg)

After:
![20220201154350_1](https://user-images.githubusercontent.com/67606569/153129910-f19dd8b2-e011-4f70-9fb4-7bfff62dc2d5.jpg)
 
with Portal 2: only the Button Prompts and the Loading Screen UI gets **B I G G E R**.
![unknown](https://user-images.githubusercontent.com/67606569/153130337-ecd735c0-6908-426a-bbee-074754d509da.png)
![unknown](https://user-images.githubusercontent.com/67606569/153130344-ba61c17d-7b36-48f9-a5e8-3d0927d9999e.png)

the rest remains the same.
